racetrack_commons/racetrack_commons/plugin/engine.py

```diff
@@ -72,11 +72,19 @@ def invoke_one_plugin_hook(self, plugin_name: str, function: Callable, *args, **
             logger.warning(f'Plugin {plugin_name} does not have hook {function_name}')
             return None
 
-    def find_plugin(self, name: str) -> PluginData:
-        for plugin_data in self.plugins_data:
-            if plugin_data.plugin_manifest.name == name:
-                return plugin_data
-        raise EntityNotFound(f'plugin named "{name}" was not found')
+    def find_plugin(self, name: str, version: Optional[str] = None) -> PluginData:
+        if version:
+            for plugin_data in self.plugins_data:
+                if plugin_data.plugin_manifest.name == name and plugin_data.plugin_manifest.version == version:
+                    return plugin_data
+            raise EntityNotFound(f'plugin named "{name}" {version} was not found')
+
+        else:
+            plugin_versions = [plugin_data for plugin_data in self.plugins_data
+                               if plugin_data.plugin_manifest.name == name]
+            if not plugin_versions:
+                raise EntityNotFound(f'plugin named "{name}" was not found')
+            return plugin_versions[-1]
```

Comment:
How do you know that the last one will be.. newest? Don't see any sort. What's the logic here anyway, return any version, or newest?

Reply:
It is supposed to be sorted by `load_plugins_from_dir` function from here: https://github.com/TheRacetrack/racetrack/blob/006fcdc136855d4f0377af46cfabf25fe93289b9/racetrack_commons/racetrack_commons/plugin/loader.py#L36
The logic is to take the newest plugin. It's sorted by "priority, then by version ascending" so that the most important plugins are invoked at the end in case they would interfere with each other. I can't find the real case now, but who knows what plugins will be created in future? 

It's not solving any problem right now, but I think it's good to have a deterministic order of execution at least. So it's rather to secure for the future.

Reply:
Ok, didn't get it from reading this function that the list will be sorted.